app/controllers/pictures_controller.rb

```diff
@@ -0,0 +1,54 @@
+class PicturesController < ApplicationController
+  before_action :authenticate_user!, :except => [:index, :show]
+
+  def index
+    @pictures = Picture.all
+  end
+
+  def create
+    @picture = current_user.pictures.new(picture_params)
+    if @picture.save
+      redirect_to pictures_path
+    else
+      redirect_to pictures_new_path
+      flash[:notice] = 'An error has occurred.'
+    end
+  end
+
+  def new
+    @picture = Picture.new
+  end
+
+  def edit
+    @picture = Picture.find(params[:id])
+    unless current_user.pictures.include? @picture
+      flash[:notice] = 'Error: You cannot edit this picture as you did not create it'
```

Comment:
Metrics/LineLength: Line is too long. [85/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)